plugins/integration/services/IntegrationService.php

```diff
@@ -48,26 +48,42 @@ public function notifyAction($id)
 	{
 		$coreType = IntegrationPlugin::getBatchJobTypeCoreValue(IntegrationBatchJobType::INTEGRATION);
 		$batchJob = BatchJobPeer::retrieveByPK($id);
-		$invalid = false;
+		$invalidJobId = false;
+		$invalidKs = false;
 		if(!$batchJob)
 		{
-			$invalid = true;
+			$invalidJobId = true;
 			KalturaLog::err("Job [$id] not found");
 		}
 		elseif($batchJob->getJobType() != $coreType)
 		{
-			$invalid = true;
+			$invalidJobId = true;
 			KalturaLog::err("Job [$id] wrong type [" . $batchJob->getJobType() . "] expected [" . $coreType . "]");
 		}
 		elseif($batchJob->getStatus() != KalturaBatchJobStatus::ALMOST_DONE)
 		{
-			$invalid = true;
+			$invalidJobId = true;
 			KalturaLog::err("Job [$id] wrong status [" . $batchJob->getStatus() . "] expected [" . KalturaBatchJobStatus::ALMOST_DONE . "]");
 		}
-		if($invalid)
+		elseif($batchJob->getPartnerId() != kCurrentContext::getCurrentPartnerId())
+		{
+			$invalidKs = true;
+			KalturaLog::err("Job [$id] of wrong partner [" . $batchJob->getPartnerId() . "] expected [" . kCurrentContext::getCurrentPartnerId() . "]");
+		}
+		elseif(!kIntegrationFlowManager::validateKs(kCurrentContext::$ks_object, $batchJob))
```

Comment:
validateKs is very specific to the API, I wouldn't implement it in the code kIntegrationFlowManager, I would keep it in the service.